lib/astutils.cpp

```diff
@@ -1162,6 +1162,14 @@ struct FwdAnalysis::Result FwdAnalysis::checkRecursive(const Token *expr, const
         if (Token::simpleMatch(tok, "goto"))
             return Result(Result::Type::BAILOUT);
 
+        if (mInInnerClass == false && Token::Match(tok, "class|struct %name% {")) {
```

Comment:
this should work.. but well I think I'd like to solve this with a recursive call. In Cppcheck, parsing inner scopes is often done by recursive calls if some special handling is needed. You can add a argument `inInnerClass` for this function. And then this code will be something like:

    if (!inInnerClass && tok->str() == "{" && tok->scope()->isClassOrStruct()) {
        FwdAnalysis::Result result = checkRecursive(expr, tok, tok->link(), exprVarIds, local, true);
        if (result != FwdAnalysis::Result::NONE)
             return result;
        tok = tok->link();
    }

Reply:
Thanks for the advice. Sounds like a better solution indeed.

Reply:
Done.